src/main/java/com/cegeka/academy/repository/ChallengeRepository.java

```diff
@@ -0,0 +1,28 @@
+package com.cegeka.academy.repository;
+
+import com.cegeka.academy.domain.Challenge;
+import com.cegeka.academy.domain.User;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public interface ChallengeRepository extends JpaRepository<Challenge, Long> {
+
+    Optional<Challenge> findOneById(Long id);
```

Comment:
Exista deja o metoda care iti aduce entitatea dupa id. Daca vrei sa iti returneze un `Optional` e in regula, daca nu vezi link.
https://docs.spring.io/spring-data/jpa/docs/current/api/org/springframework/data/jpa/repository/JpaRepository.html#getOne-ID-